src/parsing/medium.py

```diff
@@ -228,10 +228,8 @@ class Medium(AbstractMedium):
     def __init__(self, urls: Union[str, list[str]], type_fallback_urls: Optional[Union[str, list[str]]] = None):
         super().__init__()
         urls = urls if isinstance(urls, list) else [urls]
-        self.urls: list[str] = []
-        for url in urls:  # dedup, should not use a set because sequence is important
-            if url not in self.urls:
-                self.urls.append(url)
+        # dedup, should not use a set because sequence is important
+        self.urls: list[str] = [url for url in urls if url not in self.urls]
```

Comment:
Please do note that `self.urls` would not be updated during the execution of the list generator.

Try this out:
```py
l1 = list(range(10)) *2
# [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
l2 = []
l2 = [i for i in l1 if i not in l2]
# [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
```

Reply:
Well, just change it to:
```Python
self.urls: list[str] = list(dict.fromkeys(urls)
```

Reply:
I am not in favor of using `dict` here. Dictionary order do be guaranteed to be insertion order in CPython 3.7+, it is still confusing since that's not true in the past.